packages/common/src/store/purchase-content/sagas.ts

```diff
@@ -620,143 +496,31 @@ async function* purchaseAlbumWithCoinflow(args: {
-  } else {
-    throw new Error('Track is not available for purchase.')
-  }
-
-  // Check if already purchased
-  if (accessType === 'stream' && album.access?.stream) {
-    throw new Error('album already purchased')
+    albumId: encodeHashId(albumId),
+    userId: encodeHashId(userId),
+    wallet: wallet.publicKey
   }
 
-  // Check if price changed
-  if (USDC(priceNumber).value < USDC(centPrice / 100).value) {
-    throw new Error('Track price increased.')
-  }
-
-  let extraAmount = USDC(extraAmountNumber).value
-  const total = USDC(centPrice / 100.0).value + extraAmount
-  console.debug('Purchase total:', total)
-
-  // Convert splits to big int and spread extra amount to every split
-  const splits = Object.entries<bigint | number>(numberSplits).reduce(
-    (prev, [key, value], index, arr) => {
-      const amountToAdd = extraAmount / BigInt(arr.length - index)
-      extraAmount = USDC(extraAmount - amountToAdd).value
-      return {
-        ...prev,
-        [key]: BigInt(value) + amountToAdd
-      }
-    },
-    {}
-  )
-  console.debug('Calculated splits after extra amount:', splits)
-
-  // Create user bank for recipient if not exists
-  console.debug('Checking for recipient user bank...')
-  const { userBank: recipientUserBank, didExist } =
-    await sdk.services.claimableTokensClient.getOrCreateUserBank({
-      ethWallet: album.user.wallet,
-      mint: 'USDC'
-    })
-  if (!didExist) {
-    console.debug('Created user bank', {
-      recipientUserBank: recipientUserBank.toBase58()
-    })
-  } else {
-    console.debug('User bank exists', {
-      recipientUserBank: recipientUserBank.toBase58()
-    })
-  }
-
-  const audiusBackendInstance = yield* getContext('audiusBackendInstance')
-  const rootAccount = yield* call(getRootSolanaAccount, audiusBackendInstance)
-  const instructions = yield* call(
-    sdk.services.paymentRouterClient.createPurchaseContentInstructions,
-    {
-      total,
-      blockNumber: album.blocknumber,
-      buyerUserId: userId,
-      sourceWallet: rootAccount.publicKey,
-      splits,
-      mint,
-      contentType,
-      accessType,
-      contentId: albumId
-    }
-  )
   const transaction = yield* call(
-    [
-      sdk.services.paymentRouterClient,
-      sdk.services.paymentRouterClient.buildTransaction
-    ],
-    {
-      instructions
-    }
+    [sdk.tracks, sdk.albums.getPurchaseAlbumTransaction],
+    params
   )
   const serializedTransaction = Buffer.from(transaction.serialize()).toString(
     'base64'
   )
+
   const purchaseMetadata = yield* call(getCoinflowPurchaseMetadata, {
-    blocknumber: album.blocknumber,
-    purchaserUserId: userId,
-    purchaseAccess: PurchaseAccess[accessType],
     contentId: albumId,
-    contentType: PurchaseableContentType[contentType],
-    price: Number(USDC(total).toString()),
-    extraAmount: extraAmountNumber,
-    splits
+    contentType: PurchaseableContentType.ALBUM
   })
+  const total = (price + extraAmount) / 100
```

Comment:
I'm seeing this added here but not seeing it used in any of the changes?

Reply:
Used a couple lines down:
https://github.com/AudiusProject/audius-protocol/pull/8611/files#diff-05b196fd6addcb7591f9968411512d85ab9345fdebcb15dee0fc9423e64c8b83R526

It's not net new, just moved